compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -347,11 +347,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                          to invalid or uninitialized memory is undefined behavior",
                     );
                 }
-
+                let body_hir = self.tcx.hir().local_def_id_to_hir_id(self.body_id);
                 let ty_span = match actual.kind() {
-                    ty::Param(param_type) => Some(
-                        param_type.span_from_generics(self.tcx, self.body_id.owner.to_def_id()),
-                    ),
+                    ty::Param(param_type) => {
+                        Some(param_type.span_from_generics(self.tcx, body_hir.owner.to_def_id()))
+                    }
```

Comment:
same here, use `self.body_id` directly instead of going through the `HirId` owner